source/common/common/matchers.cc

```diff
@@ -118,7 +118,7 @@ PathMatcher::createSafeRegex(const envoy::type::matcher::v3::RegexMatcher& regex
 
 bool MetadataMatcher::match(const envoy::config::core::v3::Metadata& metadata) const {
   const auto& value = Envoy::Config::Metadata::metadataValue(&metadata, matcher_.filter(), path_);
-  return value_matcher_ && value_matcher_->match(value);
+  return value_matcher_ && (value_matcher_->match(value) ^ matcher_.invert());
```

Comment:
looks like `value_matcher_->match(value) ^ matcher_.invert()` is enough here.  `value_matcher_` should be always have value. anyway, it is nit.

Reply:
done.